etc/flux.service.in

```diff
@@ -16,7 +16,7 @@ ExecStart=@X_BINDIR@/flux broker \
   -Sbroker.rc2_none \
   -Sbroker.quorum=0
 ExecReload=@X_BINDIR@/flux config reload
-Restart=on-success
+Restart=always
```

Comment:
From [systemd.service(5)](https://www.freedesktop.org/software/systemd/man/systemd.service.html#Restart=):
> If set to on-success, it will be restarted only when the service process exits cleanly. In this context, a clean exit means any of the following:
>
> * exit code of 0;
> * for types other than Type=oneshot, one of the signals SIGHUP, SIGINT, SIGTERM, or SIGPIPE;
> * exit statuses and signals specified in `SuccessExitStatus=`.

Maybe we could define an exit status for "subtree panics" which could then be used with `SuccessExitStatus=` so that failing brokers are only restarted in that situation?

I'd be worried about a bad config or other fatal condition resulting in systemd restarting flux continually.

Reply:
Also, typo in the commit message
> Problem: brokers currently are not restarted by systemd if the
> exit with a non-zero return code.

"restarted by systemd if _they_"

Reply:
As discussed offline, for now i'll drop this commit as it's kind of off topic for this PR anyway, and might require further discussion.